ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -527,6 +527,13 @@ public void setAutoCapitalize(ReactEditText view, int autoCapitalize) {
 
   @ReactProp(name = "keyboardType")
   public void setKeyboardType(ReactEditText view, @Nullable String keyboardType) {
+    int passwordVisibilityFlag = InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD &
+        ~InputType.TYPE_TEXT_VARIATION_PASSWORD;
+
+    int keyboardTypeFlags = INPUT_TYPE_KEYBOARD_NUMBERED |
+            InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS |
+            InputType.TYPE_CLASS_TEXT | InputType.TYPE_CLASS_PHONE |
+            passwordVisibilityFlag;
     int flagsToSet = InputType.TYPE_CLASS_TEXT;
     if (KEYBOARD_TYPE_NUMERIC.equalsIgnoreCase(keyboardType)) {
       flagsToSet = INPUT_TYPE_KEYBOARD_NUMBERED;
```

Comment:
A bit later in this code we set the TYPE_TEXT_VARIATION_VISIBLE_PASSWORD flag, should we just set passwordVisibilityFlag instead?

Reply:
That's a good question. I think it comes down to how the field should behave when `keyboardType="visible-password"` but `secureTextEntry="false"`. Due to the way Android handles the flags, if we set the flag I called `passwordVisibilityFlag` without setting the `InputType.TYPE_TEXT_VARIATION_PASSWORD` flag, it will actually treat it like `InputType.TYPE_TEXT_VARIATION_URI`.

So three options I see:
1. Somehow forbid `keyboardType="visible-password"` with `secureTextEntry="false"`
2. Let `keyboardType="visible-password"` override the `secureTextEntry` prop (equivalent to the current PR code)
3. Allow the consumer to set the props inconsistently (probably with a warning) and risk an unexpected `InputType.TYPE_TEXT_VARIATION_URI`

I should probably leave that decision up to the core maintainers.

Reply:
Makes sense, let's keep the current behavior (current PR code).